ops/testing.py

```diff
@@ -1124,6 +1127,25 @@ def stop_services(
         for name in services:
             self._service_status[name] = pebble.ServiceStatus.INACTIVE
 
+    def restart_services(
+            self, services: typing.List[str], timeout: float = 30.0, delay: float = 0.1,
+    ) -> pebble.ChangeID:
+        # handle a common mistake of passing just a name rather than a list of names
+        if isinstance(services, str):
+            raise TypeError('restart_services should take a list of names, not just "{}"'.format(
+                services))
+        # TODO: handle invalid names
+        # Note: jam 2021-04-20 We don't implement ChangeID, but the default caller of this is
+        # Container.restart() which currently ignores the return value
```

Comment:
That is clearly a copy&paste from start_services, is there any other paths that trigger restart? Do we ever care about the change ID or expect a caller to care?

Reply:
I think the comment is still valid. I'm not sure what the value is exposing the change, but it would be useful for tracking a change across hook contexts.